internal/statestore/redis.go

```diff
@@ -80,6 +82,10 @@ func newRedis(cfg config.View) Service {
 		MaxActive:   cfg.GetInt("redis.pool.maxActive"),
 		IdleTimeout: cfg.GetDuration("redis.pool.idleTimeout"),
 		Wait:        true,
+		TestOnBorrow: func(c redis.Conn, _ time.Time) error {
```

Comment:
Is this a pattern recommended by redigo? Where where does this get invoked? In absence of this, would the next request using that redis connection fail?

Reply:
> Is this a pattern recommended by redigo?

Based on the documentation this is optional. I added this function to make sure the pool is not using bad redis connection when receiving a client call.
```
// TestOnBorrow is an optional application supplied function for checking
// the health of an idle connection before the connection is used again by
// the application. Argument t is the time that the connection was returned
// to the pool. If the function returns an error, then the connection is
// closed.
```
> Where where does this get invoked?

This is invoked when an idle connection was returned to the pool.

> In absence of this, would the next request using that redis connection fail?

Yes, although the pool will automatically remove the unhealthy connection if it hit it, in absence of this, the first call to a bad connection will always fail.

Reply:
Does this actually address any issues you are seeing - or you are taking this as a best practice?

Reply:
It resolved some of the Unavailable error I see when an idle connection got reset ( though it’s pretty rare)